src/app/rundown-view/components/rundown/rundown.component.ts

```diff
@@ -72,6 +78,24 @@ export class RundownComponent implements OnInit, OnDestroy, OnChanges {
       .catch(error => this.logger.data(error).error('Failed subscribing to rundown actions changes.'))
   }
 
+  public scrollViewToSegment(event: PartEvent): void {
+    const element: HTMLElement | null = document.getElementById(event.segmentId)
+    if (!element || this.isInsideViewport(element)) {
+      return
+    }
+
+    document.getElementById(event.segmentId)?.scrollIntoView({
+      behavior: 'smooth',
+      block: 'nearest',
+      inline: 'nearest',
+    })
+  }
+
+  public isInsideViewport(element: HTMLElement): boolean {
+    const elementRect: DOMRect = element.getBoundingClientRect()
+    return elementRect.top >= this.segmentListElement.nativeElement.clientHeight && elementRect.bottom <= (window.innerHeight || document.documentElement.clientHeight)
```

Comment:
it's most likely a problem in me, but I personally don't understand what and why we're checking here

Reply:
We don't want to scroll the view unless the element is outside of what the producer is actually looking at it. The `segmentListElement` is used because we don't want to consider the height that goes behind the header, since that isn't actually visible to the producer.